### PR TITLE
chore(deps): bump lua-resty-openssl to 1.5.1

### PR DESCRIPTION
**Pull request by bungle**
### Summary

- chore(kdf): fix the outlen type to be size_t

KAG-5095

### CODE REVIEW

File: changelog/unreleased/kong/bump-lua-resty-openssl.yml [hunk 1]
```diff
@@ -1,2 +1,2 @@
-message: "Bumped lua-resty-openssl from 1.4.0 to 1.5.0"
+message: "Bumped lua-resty-openssl from 1.4.0 to 1.5.1"
```
Comment (outsinre) [inline changelog/unreleased/kong/bump-lua-resty-openssl.yml:1]: We'd better state why this bump is required.

**Review by outsinre — APPROVED** [review]

File: changelog/unreleased/kong/bump-lua-resty-openssl.yml (hunk 1 above)
Comment (bungle) [reply changelog/unreleased/kong/bump-lua-resty-openssl.yml:1]: There is nothing worth to mention about that bump other than that we are now using the lastest, see:
https://github.com/fffonion/lua-resty-openssl/blob/master/CHANGELOG.md
https://github.com/fffonion/lua-resty-openssl/compare/1.5.0...1.5.1

Comment (bungle) [reply changelog/unreleased/kong/bump-lua-resty-openssl.yml]: I added the reason.

File: changelog/unreleased/kong/bump-lua-resty-openssl.yml [hunk 2]
```diff
@@ -1,2 +1,3 @@
-message: "Bumped lua-resty-openssl from 1.4.0 to 1.5.0"
+message: "Bumped lua-resty-openssl 1.5.1 to fix the KDF outlen type to be size_t."
```
Comment (chronolaw) [inline changelog/unreleased/kong/bump-lua-resty-openssl.yml:1]: Should we only keep the version information and remove the bug fix message?

**Review by samugi — APPROVED** [review]

**Review by chronolaw — APPROVED** [review]

**Pull request merged**